repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/keystonev2/tenantvalidation/StripTenantPrefixTest.groovy

```diff
@@ -89,9 +85,9 @@ class StripTenantPrefixTest extends ReposeValveTest {
         "1234"        | "hybrid:1234"  | "200"
         "1234"        | "foo:1234"     | "200"
         "1234"        | "bar-1234"     | "200"
+        "hybrid:1235" | "hybrid:1235"  | "200"
+        "bar-1234"    | "bar-1234"     | "200"
```

Comment:
So to start, this was a bad test. The reason it was getting a `401` was not that the user was unauthorized based on the user's tenant, but rather that the Keystone v2 filter failed to parse the tenant from the request. Specifically, this test was using the the regex `.*/accounts/(\w+)/.*` to extract the tenant from the request URI. Clearly that regex would not match the `:` in this tenant, so a `401` is returned. So we got the same status code as failed authorization, but not for the right reason.

The reason the test started failing is because we dropped support for URI tenant extraction and started using a header instead. Therefore, the regex got dropped and we started taking the tenant from the header value successfully.

That being said, if the request tenant directly matches a tenant in the response from Identity, authorization absolutely ought to pass. So that's where the `200` comes from.